File: api/tests/portscan.clj
(ns portscan
  (:gen-class)
  (:import [java.net Socket]))

(defn scan [port]
  (try
    (with-open [s (Socket. "127.0.0.1" port)]
      (println (str port " open")))
    (catch Exception _)))

(defn -main [& args]
  (doseq [port [22 80 443]]
    (scan port)))

File: api/tests/portscan.scala
import java.net.Socket
import scala.util.{Try, Using}

object portscan {
  def main(args: Array[String]): Unit = {
    for (port <- List(22, 80, 443)) {
      Try(Using.resource(new Socket("127.0.0.1", port))(_ => println(s"$port open")))
    }
  }
}
